src/constants/small_enums.rs

```diff
@@ -97,26 +35,49 @@ pub enum ErrorCode {
     GclNotEnough = -15,
 }
 
-impl From<ReturnCode> for Result<(), ErrorCode> {
-    fn from(value: ReturnCode) -> Self {
-        match value {
-            ReturnCode::Ok => Ok(()),
-            code => {
-                // SAFETY: ErrorCode is a duplicate of ReturnCode, minus the Ok variant that
-                // was covered above.
-                let err_code = unsafe { ErrorCode::from_i8(code as i8).unwrap_unchecked() };
-                Err(err_code)
-            }
+impl FromReturnCode for ErrorCode {
+    type Error = Self;
+
+    fn result_from_i8_unchecked(val: i8) -> Result<(), Self::Error> {
+        match val {
+            0 => Ok(()),
+            -1 => Err(ErrorCode::NotOwner),
+            -2 => Err(ErrorCode::NoPath),
+            -3 => Err(ErrorCode::NameExists),
+            -4 => Err(ErrorCode::Busy),
+            -5 => Err(ErrorCode::NotFound),
+            -6 => Err(ErrorCode::NotEnough),
+            -7 => Err(ErrorCode::InvalidTarget),
+            -8 => Err(ErrorCode::Full),
+            -9 => Err(ErrorCode::NotInRange),
+            -10 => Err(ErrorCode::InvalidArgs),
+            -11 => Err(ErrorCode::Tired),
+            -12 => Err(ErrorCode::NoBodypart),
+            -14 => Err(ErrorCode::RclNotEnough),
+            -15 => Err(ErrorCode::GclNotEnough),
+            // SAFETY: Return codes must always be one of the values already covered
+            _ => unsafe { std::hint::unreachable_unchecked() },
```

Comment:
Why not regular `unreachable!()` to prevent UB? Note that the trait is pub and in some circumstances, such as putting errors to some structures first, it is possible to mix them up.

Reply:
Maybe a crate feature that's off by default that turns on 'crazy enough to trust the game's return values to be what we expect' mode? Because that makes sense - we don't want this to be insanely difficult to debug in the case where we do actually get a possible return value wrong.